src/ghci/warning_formatter.rs

```diff
@@ -0,0 +1,232 @@
+//! Warning message formatting with GHC-style coloring.
+//!
+//! This module provides functionality to colorize GHC diagnostic messages
+//! to match the output format and colors used by GHC itself.
+
+use owo_colors::OwoColorize;
+use owo_colors::Stream::Stdout;
+
+use crate::ghci::parse::Severity;
+
+/// Formats warning and error messages with GHC-compatible colors.
+pub struct WarningFormatter;
```

Comment:
This has no data, so these can just be functions. You can remove this type entirely.